src/main/java/io/cdap/plugin/gcp/spanner/sink/SpannerOutputFormat.java

```diff
@@ -231,4 +243,125 @@ public void abortTask(TaskAttemptContext taskAttemptContext) {
       }
     };
   }
+
+  private static void writeArray(String name, Schema componentSchema, Object values, Mutation.WriteBuilder builder) {
+    Schema unionSchema = componentSchema.getUnionSchema(0);
+    if (unionSchema == null) {
+      throw new IllegalStateException("Union schema of '" + componentSchema.getRecordName() + "' is null");
+    }
+
+    Schema.LogicalType logicalType = unionSchema.getLogicalType();
+    if (logicalType != null) {
+      switch (logicalType) {
+        case DATE:
+          Collection<LocalDate> dateList = toCollection(name, logicalType.toString(), values);
+          builder.set(name)
+            .toDateArray(
+              dateList.stream()
+                .map(date -> Date.fromYearMonthDay(date.getYear(), date.getMonthValue(), date.getDayOfMonth()))
+                .collect(Collectors.toList())
+            );
+          break;
+        case TIMESTAMP_MILLIS:
+        case TIMESTAMP_MICROS:
+          Collection<ZonedDateTime> timestampList = toCollection(name, logicalType.toString(), values);
```

Comment:
values should only ever contain the physical representation of the value, which means you'll get back a collection of longs here, not a collection of ZonedDateTime. Similarly, date will give you a collection of ints.

Reply:
This conversion logic should all be unit tested. Please extract out the conversion logic into a new class that converts StructuredRecord into Mutation and add unit tests for it.

Reply:
ZonedDateTime. Similarly, date will give you a collection of ints.

Changed to use transformation from realization for TIMESTAMP type, made it similar to Datastore plugin and moved to RecordToEntityTransformer class. Added tests in RecordToEntityTransformerTest class